app/models/results.js

```diff
@@ -108,92 +108,78 @@ ResultSchema.plugin(QueryPlugin); // install QueryPlugin
 /**
  * Methods
  */
-ResultSchema.pre('validate', function preValidate(next) {
-  let error;
-  const buildSha1 = _.get(this, 'exec.sut.buildSha1');
-
-  const logs = _.get(this, 'exec.logs');
-  if (Array.isArray(logs)) {
-    for (let i = 0; i < logs.length; i += 1) {
-      const file = logs[i];
-      if (!file.name) {
-        error = new Error(`file[${i}].name missing`);
-        break;
-      }
-
-      if (!file.data) {
-        logger.warn(`file[${i}].data missing`);
-        file.data = '';
-        // error = new Error(`file[${i}].data missing`);
-        // break;
-      }
+ResultSchema.methods.getBuildId = function () { // eslint-disable-line func-names
+  logger.debug('lookup build..');
+  return _.get(this, 'exec.sut.ref', undefined);
+};
 
-      if (file.base64) {
-        file.data = new Buffer(file.base64, 'base64');
-        file.base64 = undefined;
-      }
+/**
+ * Validation
+ */
+function linkRelatedBuild(buildChecksum) {
+  if (!buildChecksum) {
+    return Promise.resolve();
+  }
 
-      file.size = file.data.length;
-      file.sha1 = checksum(file.data, 'sha1');
-      file.sha256 = checksum(file.data, 'sha256');
-
-      if (fileProvider === 'mongodb') {
-        // use mongodb document
-        logger.warn('store file %s to mongodb', file.name);
-      } else if (fileProvider) {
-        // store to filesystem
-        filedb.storeFile(file)
-          .then(() => {
-            logger.silly(`File ${file.name} stored`);
-          })
-          .catch((storeError) => {
-            logger.warn(storeError);
-          });
-        file.data = undefined;
-      } else {
-        // do not store at all..
-        file.data = undefined;
-        logger.warn('filedb is not configured');
-      }
+  logger.debug(`Processing result build sha1: ${buildChecksum}`);
+  return Build.findOne({'files.sha1': buildChecksum}).then((build) => {
+    if (build) {
+      logger.debug(`Build found, linking Result: ${this._id} with Build: ${build._id}`);
+      this.exec.sut.ref = build._id;
     }
-  }
+  });
+}
 
-  if (error) {
+ResultSchema.pre('validate', function (next) { // eslint-disable-line func-names
+  // Validate logs field
+  // @Todo Maybe this check is unnecessary
```

Comment:
yeah shouldn't be anything else than array - mongoose schema takes care about it i think..